zerver/webhooks/semaphore2/view.py

```diff
@@ -0,0 +1,166 @@
+# Webhooks for external integrations.
+from typing import Any, Dict
+
+from django.http import HttpRequest, HttpResponse
+
+from zerver.decorator import api_key_only_webhook_view
+from zerver.lib.request import REQ, has_request_variables
+from zerver.lib.response import json_success
+from zerver.lib.webhooks.common import check_send_webhook_message
+from zerver.models import UserProfile
+
+# There seems to be a possibility of only 2 types of events : Push or a
+# Pull request, which can trigger a pipeline.
+
+# As per the Semaphore2.0 documentation, notifications would be sent only when
+# a pipeline is triggered.
+
+PUSH_TEMPLATE = """
+**{email}** pushed to branch **{branch_name}** of [{organization_name}/{project_name}]({repository_url}):
+* **Commit**: [(#{commit_hash}) {commit_message}]({commit_url})
+* **Pipeline**:
+  - Name: {pipeline_name}
+  - State: {pipeline_state}
+  - Result: {pipeline_result}
+* **Blocks**:
+{blocks}
+""".strip()
+
+PULL_REQUEST_TEMPLATE = """
+**{email}** created a pull request to branch **{branch_name}** of [{organization_name}/{project_name}]({repository_url}):
+* **Pull Request**: [{pull_request_title}]({pull_request_url})
+* **Pipeline**:
+  - Name: {pipeline_name}
+  - State: {pipeline_state}
```

Comment:
At what point is the Pull request event triggered? Is it always when the PR is opened, in which case the state would always be the same, and probably not very useful. Similarly, for the push event too.

Reply:
I found [this](https://docs.semaphoreci.com/essentials/project-workflow-trigger-options/) page, which shows that every commit to a pull request triggers a pipeline/workflow and notifications are triggered at the completion (success or failure) of a pipeline.
I think you're right, in all the cases state will always be 'stopped'. Should i remove the state ?

Reply:
Yes, I think it can be removed.

Reply:
@punchagan I think we can use `result_reason` field for pipeline and blocks. This field will be useful to track down any issue, if the pipeline stops due to some error. There are [multiple](https://docs.semaphoreci.com/reference/pipeline-yaml-reference/#result_reason) possible values for this field which tells the reason of particular `result` value.